radiation/TwoStreamMLPEMod.F90

```diff
@@ -0,0 +1,1723 @@
+      Rd_abs = Rd_net +  Rb_net * this%band(ib)%scelb(ican,icol)%om
+
+
+      Rb_abs_leaf = (1._r8-frac_abs_snow)*Rb_abs * beam_wt_leaf / (beam_wt_leaf+beam_wt_stem)
+      Rd_abs_leaf = (1._r8-frac_abs_snow)*Rd_abs * diff_wt_leaf / (diff_wt_leaf+diff_wt_stem)
+
+      R_abs_snow = (Rb_abs+Rd_abs)*frac_abs_snow
+
+      R_abs_stem = (1._r8-frac_abs_snow)* &
+           (Rb_abs*beam_wt_stem / (beam_wt_leaf+beam_wt_stem) + &
+           Rd_abs*diff_wt_stem / (diff_wt_leaf+diff_wt_stem))
+
+
+
+
+    end associate
+    return
+  end subroutine GetAbsRad
+
+  ! ================================================================================================
+  
+  subroutine Dump(this,ib,lat,lon)
+
+    ! Dump out everything we know about these two-stream elements
+
+    class(twostream_type) :: this
+    integer,intent(in)    :: ib
+    real(r8),optional,intent(in)   :: lat
+    real(r8),optional,intent(in)   :: lon
+    integer  :: ican
+    integer  :: icol
+
+    write(log_unit,*) 'Dumping Two-stream elements for band ', ib
+    write(log_unit,*)
+    write(log_unit,*) 'rbeam atm: ',this%band(ib)%Rbeam_atm
+    write(log_unit,*) 'rdiff_atm: ',this%band(ib)%Rdiff_atm
+    write(log_unit,*) 'alb grnd diff: ',this%band(ib)%albedo_grnd_diff
+    write(log_unit,*) 'alb grnd beam: ',this%band(ib)%albedo_grnd_beam
+    write(log_unit,*) 'cosz: ',this%cosz
+    write(log_unit,*) 'snow fraction: ',this%frac_snow
+    if(present(lat)) write(log_unit,*) 'lat: ',lat
+    if(present(lon)) write(log_unit,*) 'lon: ',lon
+
+    do_can: do ican = 1,this%n_lyr
+       do_col: do icol = 1,this%n_col(ican)
+          associate(scelg => this%scelg(ican,icol), &
+                    scelb => this%band(ib)%scelb(ican,icol))
+            write(log_unit,*) '--',ican,icol,'--'
+            write(log_unit,*) 'pft:',scelg%pft
+            write(log_unit,*) 'area: ',scelg%area
+            write(log_unit,*) 'lai,sai: ',scelg%lai,scelg%sai
+            write(log_unit,*) 'Kb: ',scelg%Kb
+            write(log_unit,*) 'Kb leaf: ',scelg%Kb_leaf
+            write(log_unit,*) 'Kd: ',scelg%Kd
+            write(log_unit,*) 'Rb0: ',scelb%Rbeam0
+            write(log_unit,*) 'om: ',scelb%om
+            write(log_unit,*) 'betad: ',scelb%betad
+            write(log_unit,*) 'betab:',scelb%betab
+            write(log_unit,*) 'a: ',scelb%a
+            this%band(ib)%Rbeam_atm = 1.0_r8
+            this%band(ib)%Rdiff_atm = 1.0_r8
+            write(log_unit,*)'RDiff Down @ bottom: ',this%GetRdDn(ican,icol,ib,scelg%lai+scelg%sai)
+            write(log_unit,*)'RDiff Up @ bottom: ',this%GetRdUp(ican,icol,ib,scelg%lai+scelg%sai)
+            write(log_unit,*)'Rbeam @ bottom: ',this%GetRb(ican,icol,ib,scelg%lai+scelg%sai)
+          end associate
+       end do do_col
+    end do do_can
+
+  end subroutine Dump
+
+
+  ! ================================================================================================
+
+  subroutine ParamPrep()
```

Comment:
Minor nitpick to make the procedure more distinct from other type of parameters.
```suggestion
  subroutine RadParamPrep()
```